pandas/core/reshape/reshape.py

```diff
@@ -395,15 +395,29 @@ def pivot(self, index=None, columns=None, values=None):
     See DataFrame.pivot
     """
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        if index is None:
+            cols = [columns]
+        else:
```

Comment:
can you add a comment explaining what is going on here, ideally we would have a comment for each branch of the if/else